### ISSUE TITLE
mssql dependency is locked on the v8 major, requires v10+ to allow node versions > 18

### ISSUE THREAD

**Issue by bit-shifter-tm** [dataform-co/dataform#1617]
Hello - nodejs version 20 is now long term support, with the 18 line going into maintenance.

Through the chain of dependencies, `@dataform/cli` ends up requiring `@azure/msal-node` on the v1 line ([v2 is required to enable nodejs version 20](https://www.npmjs.com/package/@azure/msal-node#node-version-support)), detail below:

The dependency chain:

We use the latest available release of `@dataform/cli`: v2.8.0.

@dataform/cli (2.8.0) > mssql ([^8.1.2](https://github.com/dataform-co/dataform/blob/2.8.0/package.json#L87) - resolving to 8.1.4) > tedious ([^14.0.0](https://github.com/tediousjs/node-mssql/blob/v8.1.4/package.json#L34) - resolving to 14.7.0) > @azure/identity ([^2.0.4](https://github.com/tediousjs/tedious/blob/v14.7.0/package.json#L43) - resolving to 2.1.0) > @azure/msal-node (^1.10.0 - resolving to v1.18.4)

`msal-node`'s [package.lock](https://raw.githubusercontent.com/AzureAD/microsoft-authentication-library-for-js/msal-node-v1.18.4/package-lock.json) for version 1.18.4 shows 4 instances where node is limited to v18 (search for: `16 || 18"`), although I believe the only offending node engine limit is found under `"node_modules/tedious/node_modules/@azure/msal-node"` in the same file

Working backwards:

The `@azure/identity` dep. seemingly needs to be bumped to [version 3.4.0](https://github.com/Azure/azure-sdk-for-js/blob/%40azure/identity_3.4.0/sdk/identity/identity/package.json#L117) (as that's when @azure/msal-node is upped to the v2 line)

To gain `@azure/identity` at a version on or above 3.4.0, `tedious` will need to be bumped to version [16.6.1](https://github.com/tediousjs/tedious/blob/v16.6.1/package.json#L43) as that's the first time we see it.

To reference tedious at v16.6.1, `mssql` needs to updated to [v10.0.0](https://github.com/tediousjs/node-mssql/blob/v10.0.0/package.json#L34), which defines tedious at `"^16.4.0"`, which should resolve to the latest in the 16 line.

Is this something that you would be happy to receive a PR for?

All the best!

**Comment by bit-shifter-tm** [comment]
tl;dr an mssql dependency bump to `"^10.0.0"` for `@dataform/cli` should allow the use of nodejs v20 with this project.

**Comment by BenBirt** [comment]
Thanks! Do you want to send us a PR?

**Comment by bit-shifter-tm** [comment]
Sure, can do - I'll get a fork set-up.

**Comment by bit-shifter** [comment]
Revision is up
